python_modules/dagster/dagster/_core/definitions/load_asset_checks_from_modules.py

```diff
@@ -1,23 +1,29 @@
 import inspect
 from importlib import import_module
 from types import ModuleType
-from typing import Iterable, Optional, Sequence
+from typing import Iterable, Optional, Sequence, cast
 
 import dagster._check as check
+from dagster._core.definitions.assets import AssetsDefinition
 
-from .asset_checks import AssetChecksDefinition
+from .asset_checks import AssetChecksDefinition, has_only_asset_checks
 from .asset_key import (
     CoercibleToAssetKeyPrefix,
     check_opt_coercible_to_asset_key_prefix_param,
 )
-from .load_assets_from_modules import find_modules_in_package, find_objects_in_module_of_types
+from .load_assets_from_modules import (
+    find_modules_in_package,
+    find_objects_in_module_of_types,
+    prefix_assets,
+)
 
 
 def _checks_from_modules(modules: Iterable[ModuleType]) -> Sequence[AssetChecksDefinition]:
     checks = []
     for module in modules:
-        for c in find_objects_in_module_of_types(module, AssetChecksDefinition):
-            checks.append(c)
+        for c in find_objects_in_module_of_types(module, AssetsDefinition):
+            if has_only_asset_checks(c):
```

Comment:
should we make the corresponding change to `assets_from_modules`? Otherwise will this report duplicate checks for users that load assets and checks?

Reply:
This is an excellent question, I had the same thought. However this is not a problem because the existing implementation of `assets_from_modules` iterates over the keys from each `AssetsDefinition`, and those that come out of `@asset_check` don't have any keys. So `load_assets_from_modules` never returns asset checks. You can confirm by looking at `load-assets_from_modules([checks_module])` (from `test_load_asset_checks_from_modules` test).

Reply:
ah nice